### PR TITLE
[Josiris][US 1.4][FAR-154][FAR-155] SearchTreeRaoResult import/export

### PR DESCRIPTION
**Pull request by pengbo-ruidian**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x ] The commit message follows our guidelines
- [ x] Tests for the changes have been added (for bug fixes / features)

### CODE REVIEW

**Comment by baptsegu** [comment]
Hello @pengbo-ruidian, thank you for the PR !

I have two general comments : 
- could you add the import of SearchTreeRaoResults in JSON format (it could be more convenient for unit testing, and we might need it)
- It was not clear in the specification (:p), but could you test "combined" SearchTreeRaoResults json exports and imports. "combined" in the sense that the json file contains information related to SearchTreeRaoResults (exemple : computationStatus) as well as information related to the RaoComputationResults from which it extends (exemple : preContingencyResult).

See for instance : https://github.com/farao-community/farao-core/blob/master/ra-optimisation/closed-optimisation-rao/src/test/resources/closed_optimisation_rao_result.json

**Comment by pengbo-ruidian** [comment]
> 
> 
> Hello @pengbo-ruidian, thank you for the PR !
> 
> I have two general comments :
> 
>     * could you add the import of SearchTreeRaoResults in JSON format (it could be more convenient for unit testing, and we might need it)
> 
>     * It was not clear in the specification (:p), but could you test "combined" SearchTreeRaoResults json exports and imports. "combined" in the sense that the json file contains information related to SearchTreeRaoResults (exemple : computationStatus) as well as information related to the RaoComputationResults from which it extends (exemple : preContingencyResult).
> 
> 
> See for instance : https://github.com/farao-community/farao-core/blob/master/ra-optimisation/closed-optimisation-rao/src/test/resources/closed_optimisation_rao_result.json

modif done

File: ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/SearchTreeRaoResultExporter.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+     */
+    String getFormat();
+
+    /**
+     * Export a result of a remedial actions optimisation
+     *
+     * @param result The result of the remedial action optimisation
+     * @param os The output stream used for the export
+     */
+    void export(SearchTreeRaoResult result, OutputStream os) throws IOException;
```
Comment (TheMaskedTurtle) [inline ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/SearchTreeRaoResultExporter.java]: I think you can remove throws IOException

File: ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/JsonSearchTreeRaoResultExporter.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    private static final String JSON_FORMAT = "Json";
+
+    @Override
+    public String getFormat() {
+        return JSON_FORMAT;
+    }
+
+    @Override
+    public void export(SearchTreeRaoResult result, OutputStream os) throws IOException {
```
Comment (TheMaskedTurtle) [inline ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/JsonSearchTreeRaoResultExporter.java]: I think you can remove throws IOException

File: ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/SearchTreeRaoResultExporters.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+ */
+public final class SearchTreeRaoResultExporters {
+
+    private static final Supplier<List<SearchTreeRaoResultExporter>> SEARCH_TREE_RAO_RESULTS_EXPORTERS
+            = Suppliers.memoize(() -> new ServiceLoaderCache<>(SearchTreeRaoResultExporter.class).getServices())::get;
+
+    private SearchTreeRaoResultExporters() {
+    }
+
+    public static void exportSearchTreeRaoResult(SearchTreeRaoResult result, String format, OutputStream outputStream) throws IOException {
```
Comment (TheMaskedTurtle) [inline ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/SearchTreeRaoResultExporters.java]: I think you can remove throws IOException

File: ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/SearchTreeRaoResultExporter.java (hunk 1 above)
Comment (pengbo-ruidian) [reply ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/SearchTreeRaoResultExporter.java]: ok

File: ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/JsonSearchTreeRaoResultExporter.java (hunk 2 above)
Comment (pengbo-ruidian) [reply ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/JsonSearchTreeRaoResultExporter.java]: ok

File: ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/SearchTreeRaoResultExporters.java (hunk 3 above)
Comment (pengbo-ruidian) [reply ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/SearchTreeRaoResultExporters.java]: ok

File: ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/SearchTreeRaoResultExporter.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
+/**
```
[the remark this answers is not in the document]
Comment (pengbo-ruidian) [reply ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/SearchTreeRaoResultExporter.java]: ok

File: ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/SearchTreeRaoResultExporters.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
+/**
```
[the remark this answers is not in the document]
Comment (pengbo-ruidian) [reply ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/SearchTreeRaoResultExporters.java]: ok

File: ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/SearchTreeRaoResultImporters.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
+/**
```
[the remark this answers is not in the document]
Comment (pengbo-ruidian) [reply ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/SearchTreeRaoResultImporters.java]: ok

File: ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/SearchTreeRaoResultImporter.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
+/**
```
[the remark this answers is not in the document]
Comment (pengbo-ruidian) [reply ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/SearchTreeRaoResultImporter.java]: ok

**Pull request merged**